src/integrations/admin/background-indexing-integration.php

```diff
@@ -111,23 +125,32 @@ public function __construct(
 		Indexable_Indexing_Complete_Action $complete_indexation_action,
 		Post_Link_Indexing_Action $post_link_indexing_action,
 		Term_Link_Indexing_Action $term_link_indexing_action,
-		Indexing_Helper $indexing_helper
+		Indexing_Helper $indexing_helper,
+		Yoast_Admin_And_Dashboard_Conditional $yoast_admin_and_dashboard_conditional,
+		Get_Request_Conditional $get_request_conditional
 	) {
-		$this->post_indexation              = $post_indexation;
-		$this->term_indexation              = $term_indexation;
-		$this->post_type_archive_indexation = $post_type_archive_indexation;
-		$this->general_indexation           = $general_indexation;
-		$this->complete_indexation_action   = $complete_indexation_action;
-		$this->post_link_indexing_action    = $post_link_indexing_action;
-		$this->term_link_indexing_action    = $term_link_indexing_action;
-		$this->indexing_helper              = $indexing_helper;
+		$this->post_indexation                       = $post_indexation;
+		$this->term_indexation                       = $term_indexation;
+		$this->post_type_archive_indexation          = $post_type_archive_indexation;
+		$this->general_indexation                    = $general_indexation;
+		$this->complete_indexation_action            = $complete_indexation_action;
+		$this->post_link_indexing_action             = $post_link_indexing_action;
+		$this->term_link_indexing_action             = $term_link_indexing_action;
+		$this->indexing_helper                       = $indexing_helper;
+		$this->yoast_admin_and_dashboard_conditional = $yoast_admin_and_dashboard_conditional;
+		$this->get_request_conditional               = $get_request_conditional;
 	}
 
 	/**
 	 * Register hooks.
 	 */
 	public function register_hooks() {
-		\add_action( 'admin_init', [ $this, 'register_shutdown_indexing' ], 10 );
+		\add_action( 'admin_init', [ $this, 'register_shutdown_indexing' ] );
```

Comment:
With CRON indexing I'm in doubt whether or not the shutdown indexing has any added value still. I'm leaning quite strongly towards removing it as it would mean no longer having to check potentially every request on the relevant pages if this is necessary.

Reply:
I think we should keep shutdown indexing as wp-cron can be disabled. To reduce the load on admin_init on our pages, we could consider using 
`indexing_helper->is_index_up_to_date()` instead of `get_limited_filtered_unindexed_count`, which reads an autoloaded option in most cases.

Reply:
After giving it some more thought, using `is_index_up_to_date` isn't the best option. It queries the number of unindexed objects if the index wasn't complete at some point, but doesn't return the number. So we need to query it again. It's cached, but still... 
Additionally, if we don't use `is_index_up_to_date`, the shutdown indexing will kick in if (for whatever reason) there is a new unindexed object after the indexation process has been completed. We won't get that when using `is_index_up_to_date`.

@herregroen how would you feel towards only registerring the admin_init this hook if wp_cron is disabled?

Reply:
Agreed! That sounds like the best solution for now.